tests/lib/rules/no-else-return.js

```diff
@@ -182,6 +182,32 @@ ruleTester.run("no-else-return", rule, {
             output: "function foo21() { var x = true; if (x) { return x; } if (x === false) { return false; } }",
             options: [{ allowElseIf: false }],
             errors: [{ messageId: "unexpected", type: "IfStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { return bar; } else { const baz = 1;  return baz; } }",
```

Comment:
why this case cannot be fixed? IMHO, it seems safe to do the fix.

Reply:
@Aladdin-ADD what the expected output?

Reply:
@Aladdin-ADD Does my test code correct?

Reply:
This case should be fixable because the declaration in the `else` block doesn't shadow any other variable.

Reply:
It might be okay to just not autofix any block scope declarations to be safe, but I think this one *could* be autofixed since none of the declarations share a name with the variable in the immediate upper scope.